### PR TITLE
Depending on the network defined, connect to staging or prod

### PR DESCRIPTION
**Pull request by jordiamez**
If the chain is a testnet we return the staging URL, if the chain is a mainnet we return the prod URL.

### CODE REVIEW

File: packages/client/wallets/aa/src/api/CrossmintService.ts [hunk 1]
```diff
@@ -78,15 +86,26 @@ export class CrossmintService {
...
+        const jsonResponse = await response.json();
+
+        this.handleResponse(jsonResponse);
+
+        return jsonResponse;
+    }
+
+    private handleResponse(response: any) {
+        if (response && response.error) {
+            if (response.message === "API key not found") {
```
[the remark this answers is not in the document]
Comment (MatiNavCrossmint) [reply packages/client/wallets/aa/src/api/CrossmintService.ts:105]: All this file has changed, this PR needs to sync with main

File: packages/client/wallets/aa/src/api/CrossmintService.ts [hunk 2]
```diff
@@ -10,6 +13,11 @@ export class CrossmintService {
             "x-client-secret": clientSecret,
             "x-project-id": projectId,
         };
+        this.crossmintBaseUrl = "https://staging.crossmint.com/api";
+    }
+
+    setBlockchainType(blockchain: Blockchain) {
+        this.crossmintBaseUrl = getApiUrlByBlockchainType(blockchain);
     }
```
Comment (MatiNavCrossmint) [inline packages/client/wallets/aa/src/api/CrossmintService.ts:21]: I think this should be set in the constructor

File: packages/client/wallets/aa/src/CrossmintAASDK.ts [hunk 3]
```diff
@@ -67,11 +64,9 @@ export default class CrossmintAASDK {
 
             return evmAAWallet;
         } catch (e) {
-            if (e instanceof WalletSdkError) {
-                console.log(e.code, e.message);
-            } else {
-                console.error("Unexpected error:", e);
-            }
+            throw new WalletSdkError(
```
Comment (MatiNavCrossmint) [inline packages/client/wallets/aa/src/CrossmintAASDK.ts:67]: Nice

**Comment by jordiamez** [comment]
@MatiNavCrossmint - I've made the changes, thanks for the feedback! 

The line: 
`  this.crossmintService.setCrossmintUrl(chain);`
   
 can't be at the constructor, because we create the crossmintService on the init function and we do not have the CHAIN param at that point so we need to set it on the getOrCreateWallet: 
 
```
    private constructor(config: CrossmintAASDKInitParams) {
        this.crossmintService = new CrossmintService(config.clientSecret, config.projectId);
    }

    static init(params: CrossmintAASDKInitParams): CrossmintAASDK {
        return new CrossmintAASDK(params);
    }
```

**Comment by maxisch** [comment]
> @MatiNavCrossmint - I've made the changes, thanks for the feedback!
> 
> The line: ` this.crossmintService.setCrossmintUrl(chain);`
> 
> can't be at the constructor, because we create the crossmintService on the init function and we do not have the CHAIN param at that point so we need to set it on the getOrCreateWallet:
> 
> ```
>     private constructor(config: CrossmintAASDKInitParams) {
>         this.crossmintService = new CrossmintService(config.clientSecret, config.projectId);
>     }
> 
>     static init(params: CrossmintAASDKInitParams): CrossmintAASDK {
>         return new CrossmintAASDK(params);
>     }
> ```

Question here, why isn't the blockchain part of the constructor? Can (or should) the chain change during the lifecycle?

**Comment by jordiamez** [comment]
> > @MatiNavCrossmint - I've made the changes, thanks for the feedback!
> > The line: ` this.crossmintService.setCrossmintUrl(chain);`
> > can't be at the constructor, because we create the crossmintService on the init function and we do not have the CHAIN param at that point so we need to set it on the getOrCreateWallet:
> > ```
> >     private constructor(config: CrossmintAASDKInitParams) {
> >         this.crossmintService = new CrossmintService(config.clientSecret, config.projectId);
> >     }
> > 
> >     static init(params: CrossmintAASDKInitParams): CrossmintAASDK {
> >         return new CrossmintAASDK(params);
> >     }
> > ```
> 
> Question here, why isn't the blockchain part of the constructor? Can (or should) the chain change during the lifecycle?

@maxisch We have the init function on AAWallet that works as a constructor to make the connection with crossbitmain

We could add the blockchain on that point but on the document design we have only the CrossmintAASDKInitParams that includes only clientSecret projectId so we can't add it there. That's why we have the setCrossmintUrl(chain);

**Review by maxisch — APPROVED** [review]

**Pull request merged**